File: csharp/Auction-TestDesign-Kata/Test-Auction/AuctionMessageTranslatorSpyTest.cs
using Auction;

namespace Test_Auction_TestDesign_Kata;

using Moq;

public class AuctionMessageTranslatorSpyTest
{
    [Fact]
    public void NotifiesAuctionClosedWhenCloseMessageReceived()
    {
        var message = "SOLVersion: 1.1; Event: CLOSE;";
        var mockListener = new Mock<IAuctionEventListener>();
        var translator = new AuctionMessageTranslator(mockListener.Object);

        translator.ProcessMessage(message);

        mockListener.Verify(m => m.AuctionClosed());
    }    
    
    [Fact]
    public void NotifiesBidDetailsWhenPriceMessageReceived()
    {
        var message = "SOLVersion: 1.1; Event: PRICE; CurrentPrice: 192; Increment: 7; Bidder: Someone else;";
        var mockListener = new Mock<IAuctionEventListener>();
        var translator = new AuctionMessageTranslator(mockListener.Object);

        translator.ProcessMessage(message);

        mockListener.Verify(m => m.CurrentPrice(192));
    }
}

File: kotlin/src/test/kotlin/org/sammancoaching/AuctionMessageTranslatorSpyTest.kt
package org.sammancoaching

import io.mockk.mockk
import io.mockk.verify
import kotlin.test.Test

class AuctionMessageTranslatorSpyTest {

    @Test
    fun notifiesAuctionClosedWhenCloseMessageReceived() {
        val message = "SOLVersion: 1.1; Event: CLOSE;"

        val listener: AuctionEventListener = mockk(relaxed = true)
        val translator = AuctionMessageTranslator(listener)

        translator.processMessage(message)

        verify { listener.auctionClosed() }
    }

    @Test
    fun notifiesBidDetailsWhenPriceMessageReceived() {
        val message = "SOLVersion: 1.1; Event: PRICE; CurrentPrice: 192; Increment: 7; Bidder: Someone else;"
        val listener: AuctionEventListener = mockk(relaxed = true)
        val translator = AuctionMessageTranslator(listener)

        translator.processMessage(message)

        verify { listener.currentPrice(192) }
    }
}